### DISCUSSION TITLE
Unable to load Worlds on Raspberrypi

### DISCUSSION THREAD

**Discussion by Michi9409** [felddy/foundryvtt-docker#810] [Q&A] [answered]
Hey everyone,
I recently set up Foundry on my Raspberypi 4 (Raspbian V11)  and everything went smoothly until I tried to load a world. In the dev tools of my browser I found the error shown on the picture I attached. Can anyone help me resolve that problem? 

I have docker version 20.10.5+dfsg1 and docker-compose version 1.25.0 installed. 

Here is the content of the docker-compose.yml I am using: 

---
version: "3.3"

services:

  foundry:
     image: felddy/foundryvtt:release
     hostname: my-domain
     restart: "unless-stopped"
     volumes:
       - type: bind
         source: /path/to/foundrydata
         target: /data
     environment:
       - FOUNDRY_PASSWORD=my-foundry-password
       - FOUNDRY_USERNAME=my-foundry-username
       - FOUNDRY_ADMIN_KEY=my-admin-key

     ports:
      - target: 30000
        published: 30000
        protocol: tcp

Thanks

![foundry_error](https://github.com/felddy/foundryvtt-docker/assets/61057307/89735d62-2ae9-43ff-a6bb-a2caac4fed20)

**Comment by felddy** [comment]
Are you accessing the page through a proxy?   My guess is that something between you and the application is blocking the web sockets.

See:
- https://github.com/felddy/foundryvtt-docker/discussions/278#discussioncomment-2037572
- https://github.com/felddy/foundryvtt-docker/issues/118

**Reply by Michi9409** [reply]
I don't know if that rules that out, but if I reach the pi with its local IP address in my local network, I get the same error. And I did not intentionally set up any proxies.

**Comment by felddy** [comment]
Is the node process running out of memory?  Do you see anything suspicious in the container logs?  I see similar reports to yours on Discord that were the result of an out-of-memory condition.

See:
- https://discord.com/channels/170995199584108546/486930822465716249/1159246208209006804

**Reply by Michi9409** [reply]
I don't know how much memory is normally allocated to Foundry, but it seemed, that the Pi itself did not run out of memory. But I found something in the logs, apparently there is a segmentation fault occuring:

FoundryVTT | 2023-10-12 17:08:10 | [info] Software license verification succeeded
FoundryVTT | 2023-10-12 17:08:10 | [info] Server started and listening on port 30000
FoundryVTT | 2023-10-12 17:08:16 | [info] Created client session 
FoundryVTT | 2023-10-12 17:08:17 | [info] Created client session 
FoundryVTT | 2023-10-12 17:08:29 | [info] Administrator authentication successful for session 
FoundryVTT | 2023-10-12 17:08:33 | [info] Saved application configuration options to /data/Config/options.json
Segmentation fault (core dumped)
Launcher | 2023-10-12 17:08:40 | [error] Node process exited with code 139
Entrypoint | 2023-10-12 17:08:43 | [info] Starting felddy/foundryvtt container v11.311.0
Entrypoint | 2023-10-12 17:08:43 | [info] Foundry Virtual Tabletop 11.311 is installed.
Entrypoint | 2023-10-12 17:08:43 | [info] Not modifying existing installation license key.
Entrypoint | 2023-10-12 17:08:43 | [info] Setting data directory permissions.
Entrypoint | 2023-10-12 17:08:43 | [info] Starting launcher with uid:gid as foundry:foundry.
Launcher | 2023-10-12 17:08:43 | [info] Generating options.json file.
Launcher | 2023-10-12 17:08:43 | [info] Setting 'Admin Access Key'.
Launcher | 2023-10-12 17:08:44 | [info] Starting Foundry Virtual Tabletop.
FoundryVTT | 2023-10-12 17:08:48 | [info] Running on Node.js - Version 18.18.0
FoundryVTT | 2023-10-12 17:08:48 | [info] Foundry Virtual Tabletop - Version 11 Build 311
FoundryVTT | 2023-10-12 17:08:48 | [info] User Data Directory - "/data"
FoundryVTT | 2023-10-12 17:08:48 | [info] Application Options:
{
  "awsConfig": null,
  "compressSocket": false,
  "compressStatic": false,
  "fullscreen": false,
  "hostname": null,
  "hotReload": false,
  "language": "en.core",
  "localHostname": null,
  "passwordSalt": null,
  "port": 30000,
  "protocol": null,
  "proxyPort": null,
  "proxySSL": false,
  "routePrefix": null,
  "sslCert": null,
  "sslKey": null,
  "updateChannel": "stable",
  "upnp": false,
  "upnpLeaseDuration": null,
  "world": null,
  "adminPassword": "••••••••••••",
  "cssTheme": "foundry",
  "deleteNEDB": false,
  "serviceConfig": null
}

**Reply by felddy** [reply]
This looks like a memory issue.  Is the world your are loading very large?  Can you try creating a new blank world and see if that also causes a segfault?

**Reply by Michi9409** [reply]
The test world I created in the screenshot is actually a blank world with no contents. I noticed, when running 

$ sudo docker stats

I noticed in the column MEM USAGE / LIMIT, both Mem usage and limit is at all times 0B, as well as MEM % is at 0%. Did I forget to allocate memory to the container? And if so, how do I specify that?

**Comment by Michi9409** [comment, marked as answer]
I solved the problem. Apparently I had the 32 bit version of Raspbian installed, so I installed the 64 bit version, set everything up as before and it worked flawlessly. Thanks for the help!

**Reply by felddy** [reply]
I'm glad you figured it out.  I appologize for not linking to the existing issue earlier, but it seemed that you were crashing much later than others that had the 32-bit OS installed.

See:
- https://github.com/felddy/foundryvtt-docker/issues/697
